### PR TITLE
Add rocblas, cublas, and gpu-aware-mpi variants to hypre

### PR DESCRIPTION
**Pull request by jrood-nrel**
Also update some things with Umpire in hypre.

@psakievich @eugeneswalker 

@PaulMullowney maybe you could comment on needing to remove `--enable-device-memory-pool` when using Umpire?

### CODE REVIEW

File: var/spack/repos/builtin/packages/hypre/package.py [hunk 1]
```diff
@@ -112,6 +115,8 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
     conflicts("+cuda", when="+int64")
     conflicts("+rocm", when="+int64")
     conflicts("+rocm", when="@:2.20")
+    conflicts("+cublas", when="~cuda", msg="cuBLAS requires CUDA to be enabled")
+    conflicts("+rocblas", when="~rocm", msg="rocBLAS requires ROCm to be enabled")
```
Comment (psakievich) [inline var/spack/repos/builtin/packages/hypre/package.py:120]: We should also probably add a conflict for `gpu-aware-mpi` here too.

Comment (PaulMullowney) [reply var/spack/repos/builtin/packages/hypre/package.py:120]: What do you have in mind for the conflict? gpu-aware-mpi works on cuda and rocm, though you need the most recent architectures/devices on each.

Comment (psakievich) [reply var/spack/repos/builtin/packages/hypre/package.py:120]: @PaulMullowney we can add those architectures specifics as requirements. Can you provide what the minimum devices that support this feature are?

File: var/spack/repos/builtin/packages/hypre/package.py [hunk 2]
```diff
@@ -74,6 +74,9 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
     variant("fortran", default=True, description="Enables fortran bindings")
     variant("gptune", default=False, description="Add the GPTune hookup code")
     variant("umpire", default=False, description="Enable Umpire support")
+    variant("gpu-aware-mpi", default=False, description="Use GPU-aware MPI")
+    variant("rocblas", default=False, description="Use rocBLAS")
```
Comment (osborn9) [inline var/spack/repos/builtin/packages/hypre/package.py:78]: While rocblas is a valid configure option, hypre is *not* explicitly using rocBLAS so this variant may be misleading to users.

**Comment by alalazo** [comment]
What is the status if this PR?

**Comment by jrood-nrel** [comment]
I will just close it.

**Pull request closed without merging**